src/Vector.cpp

```diff
@@ -0,0 +1,335 @@
+#include <Vector.h>
+#include <stdexcept>
+
+// ======================== c-tors & d-tor =======================
+
+/**
+ * ctor creates a vector with a given size and value
+ * @param size the size of the vector
+ * @param value the value to initialize the vector with
+ **/
+Vector::Vector(size_t size, int value):
+	m_size(size),
+	m_capacity(size),
+	m_vector(new int[size])
+{
+	for (size_t i = 0; i < m_size; ++i) {
+		m_vector[i] = value; 
+	}
+}
+
+/**
+ * copy ctor creates a vector from another vector
+ * @param other the vector to copy
+ **/
+Vector::Vector(const Vector& other):
+	m_size(other.m_size),
+	m_capacity(other.m_capacity),
+	m_vector(new int[m_capacity])
+{
+	for (size_t i = 0; i < m_size && i < m_capacity; ++i) {
+		m_vector[i] = other.m_vector[i];
```

Comment:
Don't use private attributes of other like `m_vector`. This is the reason you implemented `Vector::data()` method.

Reply:
fixed